packaging/os/dnf.py

```diff
@@ -122,709 +115,245 @@
 
 '''
 
-def_qf = "%{name}-%{version}-%{release}.%{arch}"
+def init_dnf(module, repos=([], []), conf='', gpg=True):
+        obj = dnf.Base()
+        if conf:
+                if not os.path.exists(conf):
+                        err = { msg: 'Invalid configuration path',
+                                path: conf }
+                        module.fail_json(**err)
+                obj.conf.read(conf)
 
-repoquery='/usr/bin/repoquery'
-if not os.path.exists(repoquery):
-    repoquery = None
+        if not gpg:
+                obj.conf.gpgcheck = False
 
-dnfbin='/usr/bin/dnf'
+        obj.conf.assumeyes = True
 
-import syslog
+        if repos[1]:
+                for r in repos[1]:
+                        obj.repos.get_matching(r).disable()
 
-def log(msg):
-    syslog.openlog('ansible-dnf', 0, syslog.LOG_USER)
-    syslog.syslog(syslog.LOG_NOTICE, msg)
+        if repos[0]:
+                for r in repos[0]:
+                        obj.repos.get_matching(r).enable()
 
-def dnf_base(conf_file=None, cachedir=False):
+        obj.read_all_repos()
+        obj.fill_sack()
+        obj.read_comps()
 
-    my = dnf.Base()
-    my.conf.debuglevel=0
-    if conf_file and os.path.exists(conf_file):
-        my.conf.config_file_path = conf_file
-        my.conf.read()
-    my.read_all_repos()
-    my.fill_sack()
+        return obj
 
-    return my
+def ansible_result(module, rc, changed, results):
+        o = {
+                'rc' :      rc,
+                'changed':  changed,
+                'results':  results
+        }
+        module.exit_json(**o)
 
-def install_dnf_utils(module):
+def get_names(name):
+        if ',' not in name:
+                return [name]
+        return name.split(',')
```

Comment:
you should not need this function, just define names as a list in the module spec